cmake: Support coverage flags on all archs

jeremybettis opened the pull request:
Most arch's CMakeLists.txt contain rules to add compiler and linker
flags for coverage if CONFIG_COVERAGE is enabled, but 4 of them were
missing this.

Signed-off-by: Jeremy Bettis <jbettis@chromium.org>

jeremybettis committed:
cmake: Support coverage flags on all archs

Most arch's CMakeLists.txt contain rules to add compiler and linker
flags for coverage if CONFIG_COVERAGE is enabled, but 4 of them were
missing this.

Instead, set the coverage flags in arch/common/CMakeLists.txt which
affects all archs.

Signed-off-by: Jeremy Bettis <jbettis@chromium.org>

galak reviewed (changes requested):
If we are supporting this on all arch's seems like we can put this in a common place rather than duplicating it everywhere.

carlescufi commented:
> If we are supporting this on all arch's seems like we can put this in a common place rather than duplicating it everywhere.

@tejlmand could you please weigh in here?

jeremybettis commented:
> If we are supporting this on all arch's seems like we can put this in a common place rather than duplicating it everywhere.

Alright, I moved this to arch/common/CMakeLists.txt and removed the redundant copies.

jeremybettis commented:
@galak I don't know how to clear the requested changes you asked for, and merging is blocked because of that.

tejlmand reviewed (changes requested):
> > If we are supporting this on all arch's seems like we can put this in a common place rather than duplicating it everywhere.
> 
> Alright, I moved this to arch/common/CMakeLists.txt and removed the redundant copies.

Thanks, having this in a single place makes things much cleaner.

But could you please update the commit message to reflect the new changes, which is that you actually place this in a common place, so that all archs are covered.

The commit message currently says:
> Most arch's CMakeLists.txt contain rules to add compiler and linker
flags for coverage if CONFIG_COVERAGE is enabled, but 4 of them were
missing this.

ioannisg reviewed (changes requested):
As @tejlmand suggests, this needs a minor update to the commit message, indicating that the coverage flags are moved to the common ARCH place.
